finetune/input_pipeline.py

```diff
@@ -269,11 +264,14 @@ def validation_settings(self, n_examples, batch_size):
     def resampling(self, Xs, Y, context=None):
         return Xs, Y, context
 
-    def _make_dataset(self, Xs, Y, train=False, context=None, update_hook=None):
+    def _make_dataset(self, Xs, Y, train=False, context=None, update_hook=None, from_generator=False, has_targets=None):
         if Y is not None:
-            dataset = lambda: self._dataset_with_targets(Xs, Y, train=train, context=context, update_hook=update_hook)
+            has_targets = True
```

Comment:
Weird to check Y given that we're using the has_targets setting.  Is that just for the generator case?

I think long term we fix some of this code flow by keeping things bundled together until the last possible moment when we split them out.